src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,44 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final String SEPARATOR = " - ";
+    private static final String DATA_PATTERN = "dd.MM.yyyy";
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        int[] salaryInfo = new int[names.length];
+        for (String stringData : data) {
+            String[] dataEmployee = stringData.split(" ");
+            if (isDateWithinRange(dataEmployee[0], dateFrom, dateTo)) {
+                for (int i = 0; i < names.length; i++) {
+                    if (names[i].equals(dataEmployee[1])) {
+                        salaryInfo[i] += Integer.parseInt(dataEmployee[2])
+                                * Integer.parseInt(dataEmployee[3]);
+                    }
+                }
+            }
+        }
+        return getReport(names, salaryInfo, dateFrom, dateTo);
+    }
+
+    private boolean isDateWithinRange(String date, String dateFrom, String dateTo) {
+        DateTimeFormatter dateFormat = DateTimeFormatter.ofPattern(DATA_PATTERN);
+        LocalDate parsedDate = LocalDate.parse(date, dateFormat);
+        LocalDate parsedDateFrom = LocalDate.parse(dateFrom, dateFormat);
+        LocalDate parsedDateTo = LocalDate.parse(dateTo, dateFormat);
+        return !parsedDate.isAfter(parsedDateTo) && !parsedDate.isBefore(parsedDateFrom);
+    }
+
+    private String getReport(String[] name, int[] result, String dateFrom, String dateTo) {
```

Comment:
if you do not use this method and immediately generate a report in getSalaryInfo, you will be able to write all the logic in one cycle instead of two - this will affect performance